File: Problems/ambiguity_problem/Code/Python/problem.py
class Animal:
    def accion(self):
        print("miau, miau, miau")

class Herramienta:
    def accion(self):
        print("levantando vehiculo....")

class Gato(Animal, Herramienta): # (cambiar orden para observar el cambio)
    pass

mi_gato = Gato()
mi_gato.accion()  # Siempre ejecuta Animal.accion() por el orden de herencia

File: Problems/ambiguity_problem/Code/C++/problem.cpp
#include <iostream>
using namespace std;

class Animal {
public:
    void accion() {
        cout << "miau, miau, miau" << endl;
    }
};

class Herramienta {
public:
    void accion() {
        cout << "levantando vehiculo...." << endl;
    }
};

class Gato : public Animal, public Herramienta {};

int main() {
    Gato mi_gato;
    mi_gato.accion();  // Error de compilación: ambigüedad
    return 0;
}